.github/workflows/linux.yml

```diff
@@ -16,474 +16,490 @@ concurrency:
         shell: bash
-    runs-on: ubuntu-20.04-16-cores
+    runs-on: aks-linux-4-cores-16gb
+    container:
+      image: openvinogithubactions.azurecr.io/ov_build/ubuntu_22_04_x64:${{ needs.openvino_download.outputs.docker_tag }}
+      volumes: 
+        - /mount:/mount
+        - ${{ github.workspace }}:${{ github.workspace }}
+      options: -e SCCACHE_AZURE_BLOB_CONTAINER -e SCCACHE_AZURE_CONNECTION_STRING
     env:
       CMAKE_GENERATOR: Unix Makefiles
-      CMAKE_BUILD_PARALLEL_LEVEL: null
       OV_INSTALL_DIR: ${{ github.workspace }}/ov
-      CCACHE_DIR: ${{ github.workspace }}/ccache
-      CCACHE_MAXSIZE: 500Mi
-      CMAKE_CXX_COMPILER_LAUNCHER: ccache
-      CMAKE_C_COMPILER_LAUNCHER: ccache
+      INSTALL_DIR: ${{ github.workspace }}/install
+      BUILD_DIR: ${{ github.workspace }}/build
+      SRC_DIR: ${{ github.workspace }}/src
 
     steps:
       - name: Clone openvino.genai
-        uses: actions/checkout@v4
+        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
         with:
+          path: ${{ env.SRC_DIR }}
           submodules: recursive
 
-      - name: Setup Python ${{ env.PYTHON_VERSION }}
-        uses: actions/setup-python@v5
-        with:
-          python-version: ${{ env.PYTHON_VERSION }}
-          cache: 'pip'
-
       - name: Download OpenVINO package
-        uses: actions/download-artifact@v4
+        uses: actions/download-artifact@fa0a91b85d4f404e444e00e005971372dc801d16 # v4.1.8
         with:
-          name: openvino_package
+          name: ${{ needs.openvino_download.outputs.ov_artifact_name }}
           path: ${{ env.OV_INSTALL_DIR }}
-
-      - name: Extract OpenVINO packages
-        run: |
-          pushd ${OV_INSTALL_DIR}
-            tar -xzf openvino_package.tar.gz -C ${OV_INSTALL_DIR} --strip-components=1
-          popd
-
-      - name: Set apt
+          merge-multiple: true
+          
+      - name: Download Archive Distribution Package
+        uses: actions/download-artifact@fa0a91b85d4f404e444e00e005971372dc801d16 # v4.1.8
+        with:
+          name: genai_archive_${{ matrix.build-type }}
+          merge-multiple: true
+          
+      - name: Extract Artifacts
+        run: pigz -dc genai.tar.gz | tar -xf - -C ${{ env.OV_INSTALL_DIR }}
+                    
+      - name: Build Samples - ${{ matrix.build-type }}
         run: |
-          echo 'Acquire::Retries "10";' | sudo tee -a /etc/apt/apt.conf.d/80-retries > /dev/null
-          echo 'APT::Get::Assume-Yes "true";' | sudo tee -a /etc/apt/apt.conf.d/81-assume-yes > /dev/null
-          echo 'APT::Get::Fix-Broken "true";' | sudo tee -a /etc/apt/apt.conf.d/82-fix-broken > /dev/null
-          echo 'APT::Get::no-install-recommends "true";' | sudo tee -a /etc/apt/apt.conf.d/83-no-recommends > /dev/null
+          source ${{ env.OV_INSTALL_DIR }}/setupvars.sh
+          cmake -DCMAKE_BUILD_TYPE=${{ matrix.build-type }} -S ${{ env.OV_INSTALL_DIR }}/samples/cpp/ -B ${{ env.BUILD_DIR }}
```

Comment:
The samples would built as part of `genai_build_cmake` job. Why is it required to build them again separately?

Reply:
good question for authors) removed

Reply:
> for authors

I see now. The original test was verifying that genai samples can be built as part of ov package with `build_samples.sh`. And this test transformed to just a separate build.

The old test is pretty important. Can you reproduce that scenario with the new ov artifact used?

Reply:
yes, I did. So do you suggest to revert build_samples job just as a build test job?

Reply:
yes